/*
 * This file was automatically generated by EvoSuite
 * Wed Apr 18 18:19:49 GMT 2018
 */

package com.jfinal.plugin.activerecord.cache;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import com.jfinal.plugin.activerecord.cache.EhCache;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = false, useJEE = true)
public class EhCache_ESTest extends EhCache_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      EhCache ehCache0 = new EhCache();
      // Undeclared exception!
      try { 
        ehCache0.removeAll("*(O");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.jfinal.plugin.ehcache.CacheKit", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      EhCache ehCache0 = new EhCache();
      Object object0 = new Object();
      // Undeclared exception!
      try { 
        ehCache0.put("", object0, (Object) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.jfinal.plugin.ehcache.CacheKit", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      EhCache ehCache0 = new EhCache();
      // Undeclared exception!
      try { 
        ehCache0.get("+']2B}'.", (Object) "");
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.jfinal.plugin.ehcache.CacheKit", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      EhCache ehCache0 = new EhCache();
      Object object0 = new Object();
      // Undeclared exception!
      try { 
        ehCache0.remove("", object0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.jfinal.plugin.ehcache.CacheKit", e);
      }
  }
}
